nym-vpn-core/crates/nym-vpn-api-client/src/types/account.rs

```diff
@@ -57,6 +60,72 @@ fn cosmos_derivation_path() -> DerivationPath {
         .unwrap()
 }
 
+#[derive(Clone, Debug)]
+pub struct VpnApiTime {
+    // The local time on the client.
+    pub local_time: OffsetDateTime,
+
+    // The estimated time on the remote server. Based on RTT, it's not guaranteed to be accurate.
+    pub estimated_remote_time: OffsetDateTime,
+}
+
+impl VpnApiTime {
+    pub fn from_estimated_remote_time(
+        local_time: OffsetDateTime,
+        estimated_remote_time: OffsetDateTime,
+    ) -> Self {
+        Self {
+            local_time,
+            estimated_remote_time,
+        }
+    }
+
+    pub fn from_remote_timestamp(
+        local_time_before_request: OffsetDateTime,
+        remote_timestamp: OffsetDateTime,
+        local_time_after_request: OffsetDateTime,
+    ) -> Self {
+        let rtt = local_time_after_request - local_time_before_request;
+        let estimated_remote_time = remote_timestamp + (rtt / 2);
+        Self {
+            local_time: local_time_before_request,
+            estimated_remote_time,
+        }
+    }
+
+    // Local time minus remote time. Meaning if the value is positive, the local time is ahead
+    // of the remote time.
+    pub fn skew(&self) -> Duration {
+        self.local_time - self.estimated_remote_time
+    }
+
+    pub fn is_synced(&self) -> bool {
+        self.skew().abs().whole_seconds() < 30
+    }
+
+    pub fn estimate_remote_now(&self) -> OffsetDateTime {
+        tracing::debug!("Estimating remote now using skew: {}", self.skew());
+        let now = OffsetDateTime::now_utc();
+        now - self.skew()
```

Comment:
Short comment here: this is a bit dependent on the fact that we compute `skew()` by doing local - remote. If that subtraction would be inverted in the `skew` function, it would still respect its name definition, but the estimate here would be doubling the possible error.
In any case, I think this is good enough for the purposes we care about. Good job on playing with those signs and timestamps.

Reply:
I think it's a very good point. Updating the function names to try to reduce the risk of this footgun